EpsilonDriverControls/Src/DriverControls.c

```diff
@@ -232,9 +256,15 @@ void sendDriveCommandsTask(void const* arg)
         float regenPercentage = (float)getAvgRegen() / 100.0;
         float accelPercentage = (float)getAvgAccel() / 100.0;
 
-        // Read GPIO Inputs
-        forward = !HAL_GPIO_ReadPin(FORWARD_GPIO_Port, FORWARD_Pin); // `!` for active low
-        reverse = !HAL_GPIO_ReadPin(REVERSE_GPIO_Port, REVERSE_Pin);
+        // Determine vehicle direction
+        VehicleDirection vehicleDirection = determineDirection();
+
+        // Determine drive commands
+        // Forward and reverse are only asserted if the respective GPIO pin is asserted and the car is moving in that direction or is stationary
+        // If the GPIO pin is asserted, but the car is moving in the opposite direction, the output velocity and current will be set to 0, making
+        // the car coast to a stop
+        forward = !HAL_GPIO_ReadPin(FORWARD_GPIO_Port, FORWARD_Pin) && (vehicleDirection == FORWARD || vehicleDirection == STATIONARY); // `!` for active low
```

Comment:
`!HAL_GPIO_ReadPin(FORWARD_GPIO_Port, FORWARD_Pin) && (vehicleDirection == FORWARD || vehicleDirection == STATIONARY)`
I'd put that in a function, this boolean statement is getting unweildy.